cmd/detect/main.go

```diff
@@ -63,13 +63,21 @@ func runDetect(context detect.Detect) (int, error) {
 	}
 
 	if version != "" {
-		return context.Pass(buildplan.BuildPlan{
-			node.Dependency: {
-				Version:  version,
-				Metadata: buildplan.Metadata{"launch": true},
+		return context.Pass(buildplan.Plan{
+			Provides: []buildplan.Provided{
```

Comment:
Node should always be provided, not just when `.nvmrc` exists or `buildpack.yml` contains a node version. Otherwise basic NPM apps (were the node-engine-cnb provides node and npm-cnb requires it with the version from package.json) won't work.

Reply:
I'm just replicating the previous behavior.  I think what you're suggesting is a change to existing behavior and should probably be in a different issue.

Reply:
I might be missing something, but this appears to be a regression from the previous behavior.

With this change, it's no longer possible to use the node-engine-cnb without a node version specified in `.nvmrc` or `buildpack.yml`. The current behavior will provide node when the npm-cnb finds it in `package.json`, but after the PR is merged, apps will need to specify the node version in multiple places.

Reply:
I don't believe so.  Previously, if `version == ""`, it passed without any entries in the build plan (i.e. no `node.Dependency` contribution), just as it does now.

Reply:
Consider a simple node.js app with a node engine version specified in `package.json`. There is no `buildpack.yml` or `.nvmrc`.

Current buildpack implementation:
**Detect:**
node-engine-cnb: pass, does not contribute to build plan
npm-cnb: pass, reads node version from package.json and adds to build plan
**Build:**
node-engine-cnb: installs node using version from build plan
npm-cnb: installs node modules

With PR:
**Detect:**
node-engine-cnb: pass, does not contribute to build plan
npm-cnb: pass, reads node version from package.json and requires it
resolver: fails, because node.js isn't provided

Even if the npm-cnb doesn't require node.js, this buildpack is never going to provide it, so I don't think there's any way to make this work using the npm-cnb.

Reply:
Specifically, without adding a `[provide]` entries for node.js, it will always return here:
https://github.com/cloudfoundry/node-engine-cnb/pull/7/files#diff-380e79adccbaf1276bf6c91cc551cd7fR41

Whereas previously, the node-engine-cnb could rely entirely on npm-cnb to add node to the build plan.